### DISCUSSION TITLE
HEIC build issues

### DISCUSSION THREAD

**Discussion by mitch-thompson** [ImageMagick/ImageMagick#6488] [Help]
### ImageMagick version

7.1.1-12

### Operating system, version and so on

Redhat UBI-8.1

### Description

I am building a dockerfile for using ImageMagick with a java application. I've stripped down the image some to duplicate the error and posted it here: https://github.com/mitch-thompson/ImageMagickDockerfile/blob/main/Dockerfile

When I run `convert -version`  I am seeing heic listed in the delegates (built-in):

```
[root@5ec9ac33a69d ImageMagick-7.1.1-12]# convert -version
Version: ImageMagick 7.1.1-12 Q16-HDRI x86_64 b3f8ed7a7:20230625 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib fontconfig freetype heic jng jpeg png tiff zlib
Compiler: gcc (8.5)
```

If I attempt to convert a heic image to another image it is successful:
```
[root@5ec9ac33a69d ImageMagick-7.1.1-12]# convert sample1.heic sample1.jpg
[root@5ec9ac33a69d ImageMagick-7.1.1-12]# echo $?
0
[root@5ec9ac33a69d ImageMagick-7.1.1-12]# ls sample1.jpg 
sample1.jpg
```

If I go and attempt to output an image to heic format it produces an error:

```
[root@5ec9ac33a69d ImageMagick-7.1.1-12]# convert sample1.heic -auto-orient sample2.heic
convert: no encode delegate for this image format `HEIC' @ warning/constitute.c/WriteImage/1399.
convert: no encode delegate for this image format `HEIC' @ error/constitute.c/WriteImage/1408.
```

Thanks!

**Comment by fmw42** [comment]
What is your version of libheif.   You need the latest version to write to heic

**Reply by mitch-thompson** [reply]
I am using 1.16.2 of libheif and  1.0.12 of libde265.

**Comment by mitch-thompson** [comment]
I never really wound up with an answer to it, but I did wind up with a work around.  For some reason the `convert -list format|grep HEIC` was returning `HEIC  r--   High Efficiency Image Format (1.16.2)` permissions with HEIC format, so we changed a command from being `|convert - -auto-orient -` over to `|convert - -auto-orient jpg:-` for heic files since we didn't care about the output being formatted in heic.

I imagine @fmw42  is probably correct and it's something with the libheif, but I am not sure what.
